proxyserver/middleware/tempauth.go

```diff
@@ -196,20 +205,19 @@ func (ta *tempAuth) ServeHTTP(writer http.ResponseWriter, request *http.Request)
 					return false, http.StatusForbidden
 				}
 			} else {
-				// TODO: Not sure if we need a more complicated authorize func for the S3 stuff?
 				ctx.S3Auth.Account = account
-				ctx.Authorize = func(r *http.Request) (bool, int) {
-					return true, http.StatusOK
-				}
+				// Get a token for this user to be used with the rest of the request
+				request.Header.Set("X-Auth-User", key)
+				request.Header.Set("X-Auth-Key", secret)
+				_, token := ta.getToken(request.Context(), ctx, user, account, secret)
+				request.Header.Set("X-Auth-Token", token)
 			}
 		}
-		ta.next.ServeHTTP(writer, request)
-		return
 	}
 	if request.URL.Path == "/auth/v1.0" {
 		ta.handleGetToken(writer, request)
 		return
-	} else if strings.HasPrefix(request.URL.Path, "/v1") || strings.HasPrefix(request.URL.Path, "/V1") {
+	} else if ctx.S3Auth != nil || strings.HasPrefix(request.URL.Path, "/v1") || strings.HasPrefix(request.URL.Path, "/V1") {
```

Comment:
Hmm, there's a ctx nil check down a few lines from here. Probably should be moved elsewhere now, since we're derefing ctx here.

Reply:
Looks like we were derefing it earlier already.